### PR TITLE
Fix error code for the Set User command

### LINKED ISSUE #19991: [TC-DRLK-2.8] Door-lock Set User Command is returning Failure instead of  OCCUPIED.
**Problem** - Set User command with existing user index value is failed.

Actual Behavior -After giving repeated user index value we are getting Failure response

> [1656314035.358615][2757:2762] CHIP:TOO: Error: IM Error 0x00000501: General error: 0x01 (FAILURE)

Expected Behavior - After giving repeated user index value we have to get OCCUPIED

> [1656314035.358615][2757:2762] CHIP:TOO: Error: IM Error 0x00000501: General error: 0x03 (OCCUPIED)

In DUT Log (Attached), it returns "Unable to overwrite existing user [endpointId=1,userIndex=1]"

> [1656314035.338115][2974:2974] CHIP:ZCL: [createUser] Unable to overwrite existing user [endpointId=1,userIndex=1]

App used - lock app
Platform - Chip-tool - RPI-4, 8GB RAM
DUT - RPI-4, 8GB RAM
Network - Ble-wifi
Commit - 2ac3ee7

Reference:
SPEC : https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/DoorLock.adoc#434-set-user-command
Test plan : https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/door_lock_Cluster.adoc#test-procedure-8

log :
[Doorlock Set user Command logs.txt](https://github.com/project-chip/connectedhomeip/files/8989913/Doorlock.Set.user.Command.logs.txt)

### PR DESCRIPTION
**Pull request by Morozov-5F**
#### Problem
* Fixes #19991.

#### Change overview
* Fixed error code sent by the SDK to Set User command.
* Changed the YAML `DL_UsersAndCredentials` test. 

#### Testing
* Verified that the door lock sends the Occupied response when Set User command is issues multiple times to the same slot.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/app/tests/suites/DL_UsersAndCredentials.yaml [hunk 1]
```diff
@@ -152,8 +152,7 @@ tests:
               - name: "credentialRule"
                 value: null
       response:
-          # So far there's no way to verify cluster-specific error codes so we're just checking generic failure here
-          error: 0x01
+          error: 0x03
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/DL_UsersAndCredentials.yaml:155]: No, the error will in fact be `0x01`.

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 2]
```diff
@@ -1659,8 +1659,7 @@ EmberAfStatus DoorLockServer::createUser(chip::EndpointId endpointId, chip::Fabr
     {
         emberAfDoorLockClusterPrintln("[createUser] Unable to overwrite existing user [endpointId=%d,userIndex=%d]", endpointId,
                                       userIndex);
-        // TODO: Add cluster-specific failure of DlStatus::kOccupied
-        return EMBER_ZCL_STATUS_FAILURE;
+        return static_cast<EmberAfStatus>(DlStatus::kOccupied);
```
Comment (bzbarsky-apple) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:1662]: This is not right.  This is going to send this value as a _global_ status.  But this is a cluster-specific status, no?

It needs to be encoded differently than a global status on the wire, and hence we need to tell the CommandHandler which one we are sending.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 2 above)
Comment (Morozov-5F) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:1662]: Got it. Is there any example on how to return the cluster-specific code?

**Review by bzbarsky-apple — APPROVED** [review]
Looks great, thanks!   You probably want to install clang-format-11 (or 12, or 13, something before 14) to fix the restyled issues.  you are getting in generated code....

**Pull request merged**